### PR TITLE
Dao implementation

### PR DESCRIPTION
**Pull request by bing-j**
Code associated with #15. Note that no tests have been written yet.

### CODE REVIEW

File: src/main/java/data_access/DummyDataAccessObject.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
+package data_access;
+
+import use_case.add_project.AddProjectDataAccessInterface;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * This is the DAO that will be used in unit testing in the beninging.
```
Comment (wojiaotreaty) [inline src/main/java/data_access/DummyDataAccessObject.java:10]: beginning*

Comment (BraedenChoo) [reply src/main/java/data_access/DummyDataAccessObject.java:10]: Rocket ship incoming

Comment (wojiaotreaty) [reply src/main/java/data_access/DummyDataAccessObject.java:10]: Rocket ship incoming

Comment (skrtsamjang) [reply src/main/java/data_access/DummyDataAccessObject.java:10]: Rocket ship incoming

Comment (VedantTheGreat) [reply src/main/java/data_access/DummyDataAccessObject.java:10]: Rocket ship incoming

File: src/main/java/data_access/DummyDataAccessObject.java
Comment (wojiaotreaty) [inline src/main/java/data_access/DummyDataAccessObject.java:1]: Does this need to be pushed to main?

File: src/main/java/data_access/UsersDataAccessObject.java
Comment (wojiaotreaty) [inline src/main/java/data_access/UsersDataAccessObject.java:1]: Please add a method with the following signature: 

String generateNewProjectId(); 

Which generates a unique project ID. This method is defined in the AddProjectDataAccessInterface.

File: src/main/java/data_access/UsersDataAccessObject.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                writer.newLine();
+            }
+        }
+    }
+
+    public void setCurrentUser(User currentUser) {
+        this.currentUser = currentUser;
+    }
+
+    public User getCurrentUser(){
```
Comment (wojiaotreaty) [inline src/main/java/data_access/UsersDataAccessObject.java:57]: We no longer need this method as no DAO Interfaces rely on it.

File: src/main/java/interface_adapter/Controller.java
Comment (wojiaotreaty) [inline src/main/java/interface_adapter/Controller.java:1]: Unsure if this file will cause a merge conflict. Is it possible to remove this?

File: src/main/java/use_case/UseCase.java
Comment (wojiaotreaty) [inline src/main/java/use_case/UseCase.java:1]: Unsure if this file will cause a merge conflict. Is it possible to remove this?

File: src/main/java/use_case/add_project/AddProjectDataAccessInterface.java
Comment (wojiaotreaty) [inline src/main/java/use_case/add_project/AddProjectDataAccessInterface.java:1]: This DAO interface will cause a merge conflict. This interface will be pushed by the branch responsible for the AddProject use case #17. Please remove this.

**Review by wojiaotreaty — CHANGES_REQUESTED** [review]
Everything looks good! Thank you for your hard work. I've left a few comments, mainly suggestions but there are a few changes that must be made before pulling this branch to main. Please take a look at the comments!

**Review by wojiaotreaty — APPROVED** [review]
Changes were made and the code looks good now!

**Pull request merged**